src/agent/multitype_queue/src/persistence_factory.hpp

```diff
@@ -7,8 +7,13 @@
 #include <string>
 #include <vector>
 
+enum PersistenceType
```

Comment:
Can this be an `enum class`?

Also, can it be nested inside the public part of `PersistenceFactory` ?